### PR TITLE
Task 5.1 implement book order dao

### PR DESCRIPTION
**Pull request by AleksandrRadchenko**

### CODE REVIEW

File: src/main/java/com/epam/wl/dao/BookOrderDAO.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+     * @return 1 (number of rows changed) if success, or throws SQLexception
+     * @throws SQLException
+     */
+    @SuppressWarnings("JavaDoc")
+    int create(
+            final BookInstance bookInstance,
+            final UserOrder userOrder,
+            final BookOption bookOption)
+            throws SQLException {
+        String updateQuery = "INSERT INTO book_order (book_instanceid, user_orderid, option) VALUES(%d, %d, '%s');";
```
Comment (ginsenghillock) [inline src/main/java/com/epam/wl/dao/BookOrderDAO.java]: Make all strings, that you use like queries, private static final, it'll be better for performance

File: src/main/java/com/epam/wl/dao/BookOrderDAO.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+     * @throws SQLException
+     */
+    @SuppressWarnings("JavaDoc")
+    public int update(BookOrder newBookOrder) throws SQLException {
+        final String updateQuery = "UPDATE book_order SET book_instanceid=%d, user_orderid=%d, option='%s'";
+        executor.executeUpdate(String.format(updateQuery,
+                newBookOrder.getBookInstanceId(),
+                newBookOrder.getOrderId(),
+                newBookOrder.getBookOption().toString()));
+        return 1;
```
Comment (TimWinnicheck) [inline src/main/java/com/epam/wl/dao/BookOrderDAO.java]: Can't this and other methods be void and not return 1?

**Comment by vlshat** [comment]
Why don't you use prepared statement?

File: src/main/java/com/epam/wl/dao/BookOrderDAO.java (hunk 1 above)
Comment (AleksandrRadchenko) [reply src/main/java/com/epam/wl/dao/BookOrderDAO.java]: Done

File: src/main/java/com/epam/wl/dao/BookOrderDAO.java (hunk 2 above)
Comment (AleksandrRadchenko) [reply src/main/java/com/epam/wl/dao/BookOrderDAO.java]: Done

**Comment by AleksandrRadchenko** [comment]
Make BookOrderDAO work with new executor (via PreparedStatements, TASK-6.6, not merged in master yet)

**Comment by TimWinnicheck** [comment]
Reviewed. Resolve merge conflicts and merge the task.

**Pull request merged**